UnitTests/FileDialogTests.cs

```diff
@@ -0,0 +1,319 @@
+			Assert.True (dlg.Canceled);
+			Assert.Equal ("x", Path.GetFileName (dlg.Path));
+
+			// complete auto typing
+			Send ('\t', ConsoleKey.Tab, false);
+
+			// but do not close dialog
+			Assert.True (dlg.Canceled);
+			Assert.EndsWith ("xx" + Path.DirectorySeparatorChar, dlg.Path);
+
+			// press enter again to confirm the dialog
+			Send ('\n', ConsoleKey.Enter, false);
+			Assert.False (dlg.Canceled);
+			Assert.EndsWith ("xx" + Path.DirectorySeparatorChar, dlg.Path);
+		}
+
+		[Fact, AutoInitShutdown]
+		public void DoNotConfirmSelectionWhenFindFocused ()
+		{
+			var dlg = GetInitializedFileDialog ();
+			var openIn = Path.Combine (Environment.CurrentDirectory, "zz");
+			Directory.CreateDirectory (openIn);
+			dlg.Path = openIn + Path.DirectorySeparatorChar;
+
+			Send ('f',ConsoleKey.F,false,false,true);
+
+			Assert.IsType<TextField> (dlg.MostFocused);
+			var tf = (TextField) dlg.MostFocused;
+			Assert.Equal ("Enter Search", tf.Caption);
+
+			// Dialog has not yet been confirmed with a choice
+			Assert.True (dlg.Canceled);
+
+			//pressing enter while search focused should not confirm path
+			Send ('\n', ConsoleKey.Enter, false);
+
+			Assert.True (dlg.Canceled);
+
+			// tabbing out of search 
+			Send ('\t', ConsoleKey.Tab, false);
+
+			//should allow enter to confirm path
+			Send ('\n', ConsoleKey.Enter, false);
+
+			// Dialog has not yet been confirmed with a choice
+			Assert.False(dlg.Canceled);
+		}
+
+		[Fact, AutoInitShutdown]
+		public void TestDirectoryContents_Linux()
+		{
+			if(System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+			{
+				// Cannot run test except on linux :( 
+				// See: https://github.com/TestableIO/System.IO.Abstractions/issues/800
+				return;
+			}
+
+			// Arrange
+			var fileSystem = new MockFileSystem(new Dictionary<string, MockFileData>(),"/");
+			fileSystem.MockTime(()=>new DateTime(2010,01,01,11,12,43));
+
+			fileSystem.AddFile( @"/myfile.txt", new MockFileData("Testing is meh."){LastWriteTime = new DateTime(2001,01,01,11,12,11)});
+			fileSystem.AddFile( @"/demo/jQuery.js", new MockFileData("some js"){LastWriteTime = new DateTime(2001,01,01,11,44,42)});
+			fileSystem.AddFile( @"/demo/image.gif", new MockFileData(new byte[] { 0x12, 0x34, 0x56, 0xd2 }){LastWriteTime = new DateTime(2002,01,01,22,42,10)});
+			
+			var m = (MockDirectoryInfo)fileSystem.DirectoryInfo.New(@"/demo/subfolder");
+			m.Create();
+			m.LastWriteTime = new DateTime(2002,01,01,22,42,10);
+			
+			fileSystem.AddFile( @"/demo/subfolder/image2.gif", new MockFileData(new byte[] { 0x12, 0x34, 0x56, 0xd2 }){LastWriteTime = new DateTime(2002,01,01,22,42,10)});
+
+			var fd = new FileDialog(fileSystem){
+				Height = 15
+			};
+			fd.Path = @"/demo/";
+			Begin(fd);
+
+			fd.Redraw(fd.Bounds);
+
+			fd.Style.DateFormat = "yyyy-MM-dd hh:mm:ss";
+
+			string expected = 
+			@"
+ ┌┤OPEN├────────────────────────────────────────────────────────────┐
```

Comment:
Remove setting of `Title` from any test where you're not explicitly testing Title functionality. We're going to evolve the look and feel of how Title is drawn and this way we won't have to mess with so many unit tests.

Reply:
Fixed in [0189a38](https://github.com/gui-cs/Terminal.Gui/pull/2259/commits/0189a3836d6ff1d8d4c7177177665981116a6210)